plugins/module_utils/oidc_auth.py

```
@@ -0,0 +1,150 @@
+
+import json
+from typing import Any, Dict, Optional, Tuple
+from urllib.error import HTTPError
+from urllib.parse import urlencode
+
+from ansible.module_utils.urls import open_url
+
+
+def http_get_json(url: str, verify_ssl: bool, ca_path: Optional[str], request_timeout: Optional[float]) -> Dict[str, Any]:
+    response = open_url(url, method='GET', validate_certs=verify_ssl, ca_path=ca_path,
+                         timeout=request_timeout or 30)
```

Flake8 (7.3.0)

[error] continuation line over-indented for visual indent

(E127)